controller/tokenization_manager.py

```diff
@@ -0,0 +1,206 @@
+            project_id, task_id, initial_count
+        )
+        text_attributes = attribute.get_text_attributes(project_id).keys()
+        non_text_attributes = attribute.get_non_text_attributes(project_id).keys()
+
+        full_count = record.count_records_without_tokenization(project_id)
+        chunk_size, progress_per_chunk = __get_chunk_size_and_progress_per_chunk(
+            full_count
+        )
+        records = record.get_records_without_tokenization(project_id)
+        chunks = [
+            records[x : x + chunk_size] for x in range(0, len(records), chunk_size)
+        ]
+        for idx, record_chunk in enumerate(chunks):
+            entries = []
+            for record_item in record_chunk:
+                if __remove_from_priority_queue(project_id, record_item.id):
+                    continue
+                entries.append(
+                    tokenize_record(project_id, tokenizer, record_item, text_attributes)
+                )
+            general.add_all(entries)
+            upload_to_minio_after_every_10th_chunk(idx, project_id, non_text_attributes)
+            update_tokenization_progress(
+                project_id, tokenization_task, progress_per_chunk
+            )
+        finalize_task(project_id, user_id, non_text_attributes, tokenization_task)
+    except Exception:
+        __handle_error(project_id, user_id, task_id)
+    finally:
+        __wrap_up_tokenization()
+
+
+def manage_record_tokenization(project_id: str, record_id: str) -> int:
+    if record.has_byte_data(project_id, record_id):
+        return 200
+    try:
+        __add_to_priority_queue(project_id, record_id)
+
+        text_attributes = attribute.get_text_attributes(
+            project_id,
+            state_filter=[
+                enums.AttributeState.UPLOADED.value,
+                enums.AttributeState.USABLE.value,
+                enums.AttributeState.RUNNING.value,
+            ],
+        )
+        tokenizer = get_tokenizer_by_project(project_id)
+        record_item = record.get(project_id, record_id)
+        entry = tokenize_record(
+            project_id, tokenizer, record_item, text_attributes, update_statistic=True
+        )
+        general.add(entry)
+        general.commit()
+        return 200
+    except Exception:
+        __remove_from_priority_queue(project_id, record_id)
+        print(traceback.format_exc(), flush=True)
+        return 418
+
+
+def __get_value_ids_string_for_update(values: List[Dict[str, Any]]) -> str:
+    value_ids = [f"'{value.get('_id')}'" for value in values]
+    value_ids = ", ".join(value_ids)
+    return "(" + value_ids + ")"
+
+
+def __get_chunk_size_and_progress_per_chunk(full_count: int) -> Tuple[int, float]:
+    chunk_size = 500
+    progress_per_chunk = round(chunk_size / full_count, 3)
+    return chunk_size, progress_per_chunk
+
+
+def __check_attribute_is_text(attribute_item: Attribute) -> None:
+    if attribute_item.data_type != enums.DataTypes.TEXT.value:
```

Comment:
maybe also include check for none of attriubte_item if the query doesn't return anything (for whatever reason :D)

Reply:
doesnt seem necessary for me tbh, why would i check it here?

Reply:
not the data_type would be missing but the requested attribute_id -> this would result in the attribute_item being a None Type. This would run into an error since NoneType doesn't have the attribute data_type which the code tries to access

Reply:
added check